Doc/library/sqlite3.rst

```diff
@@ -601,25 +601,15 @@ Connection objects
 
       Example:
 
-      .. testcode::
-
-         def dict_factory(cursor, row):
-             d = {}
-             for idx, col in enumerate(cursor.description):
-                 d[col[0]] = row[idx]
-             return d
-
-         con = sqlite3.connect(":memory:")
-         con.row_factory = dict_factory
-         cur = con.execute("SELECT 1 AS a")
-         print(cur.fetchone()["a"])
-
-         con.close()
-
-      .. testoutput::
-         :hide:
+      .. doctest::
 
-         1
+         >>> def dict_factory(cursor, row):
+         ...     return {col[0]: row[i] for i, col in enumerate(cursor.description)}
+         >>> con = sqlite3.connect(":memory:")
+         >>> con.row_factory = dict_factory
+         >>> for row in con.execute("SELECT 1 as a, 2 as b"):
+         ...     print(row)
```

Comment:
Why using a `for` if there's only one row?

Reply:
Because I want to promote the cursor iterator wherever possible :)

Reply:
IOW, I'd like to keep it that way, unless you find it very confusing. Alternatively, we could set up a test table with multiple columns and query that, but that will expand the example and possibly be distracting, as we want to focus on the row factory here.

Reply:
Fair enough.